### DISCUSSION TITLE
The link is not getting the proper straight line when I change it from horizontal to vertical

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2213] [Q&A] [answered]
### Introduction

I am using the orthogonal router for the link
The link is not getting the proper straight line when I change it from horizontal to vertical
how can I solve that issue?
![grid](https://github.com/clientIO/joint/assets/117301232/9ddf3912-5ee9-4bfc-9833-e2fd79b9609a)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
- a) Make sure the source point also falls into the grid (when you drag the target arrowhead it is always snapped to the current paper's grid). I am not sure how you created the link in the first place.
- b) Use the [snapLinksSelf](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Paper.prototype.options.snapLinksSelf) option.

**Reply by reshin-pulapaddy** [reply]
Thank you, it is working

**Reply by reshin-pulapaddy** [reply]
@kumilingus 
When I add snapLinksSelf: true to the paper, I cannot connect the link to other elements
how to solve this issue?
![snapLinksSelf](https://github.com/clientIO/joint/assets/117301232/08d0ad0f-ccf2-4733-862a-c5bebdcad6c3)

**Reply by kumilingus** [reply]
It was a bug in JointJS. Fixed by https://github.com/clientIO/joint/pull/2236. To be released today or Monday.

**Reply by reshin-pulapaddy** [reply]
I am using the 3.6.3 version
Would it be possible to get the above fix without updating the version?

**Reply by kumilingus** [reply]
You can patch your current version.
```js
joint.dia.LinkView.prototype._snapArrowheadSelf = function(evt, x, y) {
   /* patched code from #2236 */
}
```

**Reply by reshin-pulapaddy** [reply]
Thank you @kumilingus
